components/mpas-ocean/src/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -543,27 +554,42 @@ subroutine ocn_tracer_hmix_Redi_init(domain, err)!{{{
             call mpas_dmpar_finalize(domain%dminfo)
          end if
 
-         ! Add resolution taper
-         if (config_eddying_resolution_taper == 'none') then
-            ! Nothing to do, as we just keep the same assignment as above.
-         else if (config_eddying_resolution_taper == 'ramp') then
+         ! Initialize horizontal taper
+         if (config_Redi_horizontal_taper == 'none' .or. &
+             config_Redi_horizontal_taper == 'RossbyRadius') then
+            ! For 'RossbyRadius', the taper is recomputed at every time step in
+            ! the ocn_GM_compute_Bolus_velocity subroutine
+            !$omp parallel
+            !$omp do schedule(runtime)
+            do iEdge = 1,nEdges
+               RediHorizontalTaper(iEdge) = 1.0_RKIND
+            end do
+            !$omp end do
+            !$omp end parallel
+         else if (config_Redi_horizontal_taper == 'ramp') then
+            coef = 1.0_RKIND &
+                    /(config_Redi_horizontal_ramp_max - config_Redi_horizontal_ramp_min)
             !$omp parallel
             !$omp do schedule(runtime) 
             do iEdge=1,nEdges
-               if (dcEdge(iEdge) <= config_eddying_resolution_ramp_min) then
+               if (dcEdge(iEdge) <= config_Redi_horizontal_ramp_min) then
                   RediKappa(iEdge) = 0.0_RKIND
-               else if (dcEdge(iEdge) >= config_eddying_resolution_ramp_max) then
+                  RediHorizontalTaper(iEdge) = 0.0_RKIND
+               else if (dcEdge(iEdge) >= config_Redi_horizontal_ramp_max) then
                   ! nothing to do, i.e. RediKappa(iCell) = RediKappa(iCell)
+                  RediHorizontalTaper(iEdge) = 1.0_RKIND
                else
                   RediKappa(iEdge) = RediKappa(iEdge) &
-                                     *(dcEdge(iEdge) - config_eddying_resolution_ramp_min) &
-                                     /(config_eddying_resolution_ramp_max - config_eddying_resolution_ramp_min)
+                                     *(dcEdge(iEdge) - config_Redi_horizontal_ramp_min) &
+                                     /(config_Redi_horizontal_ramp_max - config_Redi_horizontal_ramp_min)
```

Comment:
This fixed the BFB mismatch. I had to put this block of code in the same order as before, with the operations in the same order as before. Now it is BFB in E3SM tests (it was BFB in stand-alone tests before, which is partly what took me so long to find it).

Reply:
yay, @mark-petersen, that's great!